spark3/src/test/java/org/apache/iceberg/spark/sql/TestSelect.java

```diff
@@ -49,8 +49,8 @@ public TestSelect(String catalogName, String implementation, Map<String, String>
 
```

Comment:
I think we should also try to test it for spark2, maybe update some tests in `TestReadProjection`?

Reply:
Yeah I actually also spent some time on this but wasn't able to find a good place to add in spark2, and later gave up thinking that the added logic was relatively simple anyway. To me `TestReadProjection` is more about testing projection which is not what we are doing. I guess I'll create a `TestSelect` in spark3 test suite and duplicate this class then.

Reply:
Added a `TestSelect` in spark2 by basically duplicating logic from the same class in spark3; although apart from basic sanity testing I'm not sure how helpful the tests are as some of the logic for examine pushed-down filters only exist in spark3...